### PR TITLE
fix(router): #1338, #1339

### PR DESCRIPTION
**Pull request by Sayan751**
# Pull Request

### CODE REVIEW

**Review by Vheissu — APPROVED** [review]

File: docs/user-docs/developer-guides/routing/navigating.md [hunk 1]
```diff
@@ -105,10 +105,11 @@ And where things really start to get interesting is when you want to pass parame
 <a load="route:profile; params.bind:{name: 'rob'}">View Profile</a>
 ```
 
-In the above example, we provide the route value. But, then also provide an object of parameters. These parameter values correspond to any parameters configured in your route definition. In our case, our route looks like this:
+In the above example, we provide the route (`id`) value. But, then also provide an object of parameters. These parameter values correspond to any parameters configured in your route definition. In our case, our route looks like this:
```
Comment (bigopon) [inline docs/user-docs/developer-guides/routing/navigating.md:108]:
```suggestion
In the above example, we provide the route (`id`) value via `route: profile`. But, then also provide an object of parameters via `params.bind: { name: 'rob' }`. These parameter values correspond to any parameters configured in your route definition. In our case, our route looks like this:
```

File: packages/__tests__/router/smoke-tests.spec.ts [hunk 2]
```diff
@@ -988,5 +991,32 @@ describe('router (smoke tests)', function () {
 
       await au.stop();
     });
+
+    it('shows title correctly', async function () {
```
Comment (bigopon) [inline packages/__tests__/router/smoke-tests.spec.ts:995]: 👍

File: packages/router/src/router.ts [hunk 3]
```diff
@@ -106,6 +106,13 @@ export class RouterOptions {
      * Default: `ignore`
      */
     public readonly sameUrlStrategy: ValueOrFunc<SameUrlStrategy>,
+
+    /**
+     * An optional handler to build the tile.
+     * When configured, the work of building the title string is completely handed over to this function.
+     * If this function returns `null`, the title is not updated.
+     */
+    public readonly buildTitle: ((transition: Transition) => string | null) | null,
```
Comment (bigopon) [inline packages/router/src/router.ts:115]: 👍

File: packages/router/src/router.ts [hunk 4]
```diff
@@ -426,6 +436,9 @@ export class Router {
   private nextTr: Transition | null = null;
   private locationChangeSubscription: IDisposable | null = null;
 
+  /** @internal */
+  public readonly hasTitleBuilder: boolean = false;
```
Comment (bigopon) [inline packages/router/src/router.ts:440]: shouldn't this be a private prop? And start with `_` to minify?

**Review by bigopon — APPROVED** [review]
Great find! 👍

File: packages/router/src/router.ts (hunk 4 above)
Comment (Sayan751) [reply packages/router/src/router.ts]: You are right. I wanted this to be `private` too. Moreover, I also wanted to express that this is a `readonly` property. However, this property cannot be set in constructor, rather inside the `Router#start` method as follows:

```typescript
(this as Writable<Router>)._hasTitleBuilder = typeof this.options.buildTitle === 'function';
```

And the `Writable`/TS doesn't pick up the private properties. However, I have prefixed the prop with `_` and that does the trick of minifying the property name.

**Pull request merged**